addons/base_import/models/base_import.py

```diff
@@ -1042,12 +1042,15 @@ def _convert_import_data(self, fields, options):
 
         if options.get('has_headers'):
             rows_to_import = rows_to_import[1:]
-        data = [
-            list(row) for row in map(mapper, rows_to_import)
-            # don't try inserting completely empty rows (e.g. from
-            # filtering out o2m fields)
-            if any(row)
-        ]
+        try:
+            data = [
+                list(row) for row in map(mapper, rows_to_import)
+                # don't try inserting completely empty rows (e.g. from
+                # filtering out o2m fields)
+                if any(row)
+            ]
+        except Exception:
+            raise ImportValidationError(_("Invalid formatting"))
```

Comment:
Maybe add a hint to the error message about invalid separator?

Reply:
> Maybe add a hint to the error message about invalid separator?

This error is generated due to the wrong separator or text delimiter. So can we add an Error hint like `invalid separator or text delimiter`?

Reply:
It's not obvious that `except Exception` catches only Invalid separator error. You should justify it or implement another exception or check

Reply:
> It's not obvious that `except Exception` catches only Invalid separator error. You should justify it or implement another exception or check

Changes are done!